rope/base/oi/doa.py

```diff
@@ -114,24 +118,55 @@ class _SocketReceiver(_MessageReceiver):
     def __init__(self):
         self.server_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
         self.data_port = 3037
+        self.key = uuid.uuid4().bytes
```

Comment:
could just use `os.urandom(16)` and it'd be more obviously correct. (It's not obvious without reading its source that uuid4 uses a CSPRNG, although it "should", and does.)

Reply:
In fact, I think usually people prefer to use the digest size as the key size, so 16 is half as big as it should be. I'd replace with `os.urandom(32)`. (Without making any statement about whether 128 bits of security is too few or whatever). Trying to remember where I heard this but can't find it.

Reply:
Much nicer, thank you.